internal/component/otelcol/exporter/datadog/datadog.go

```diff
@@ -0,0 +1,127 @@
+// Package datadog provides an otelcol.exporter.datadog component.
+// Maintainers for the Grafana Alloy wrapper:
+//	- @polyrain
+
+package datadog
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/grafana/alloy/internal/component"
+	"github.com/grafana/alloy/internal/component/otelcol"
+	otelcolCfg "github.com/grafana/alloy/internal/component/otelcol/config"
+	"github.com/grafana/alloy/internal/component/otelcol/exporter"
+	datadog_config "github.com/grafana/alloy/internal/component/otelcol/exporter/datadog/config"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+const (
+	DATADOG_TRACE_ENDPOINT   = "https://trace.agent.%s"
+	DATADOG_METRICS_ENDPOINT = "https://api.%s"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "otelcol.exporter.datadog",
+		Community: true,
+		Args:      Arguments{},
+		Exports:   otelcol.ConsumerExports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := datadogexporter.NewFactory()
+			return exporter.New(opts, fact, args.(Arguments), exporter.TypeAll)
+		},
+	})
+}
+
+// Arguments configures the otelcol.exporter.datadog component.
+type Arguments struct {
+	Client otelcol.HTTPClientArguments `alloy:"client,block,optional"`
```

Comment:
As discussed in slack, it seems that this type is not a good fit because some attributes are not supported, including the `endpoint` one which is required. I would recommend creating a custom type to solve the problem and make the component's usage clearer